### PR TITLE
feat(schedule flags): add scheduled changes API

### PR DESCRIPTION
**Pull request by jurajmajerik**
## Changes
Add the `scheduled_changes` API.

- This API populates the `posthog_scheduledchange` table, which stores the list of changes to be completed.
- The `posthog_scheduledchange` table will be processed by a Celery task which will populate the Feature Flag with new changes (TODO).
- In the future, this table will allow us to schedule changes to other models too.
- UI changes to follow in a separate PR.

## How did you test this code?
API test, 👀

### CODE REVIEW

**Review by liyiy — APPROVED** [review]
looks good, just the above feedback. Not sure if you want @neilkakkar to pop in for feedback before merging, so up to you 😆

File: posthog/api/test/test_scheduled_change.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+from rest_framework import status
+from posthog.test.base import (
+    APIBaseTest,
+)
+from posthog.models import ScheduledChange
+
+
+class TestScheduledChange(APIBaseTest):
```
[the remark this answers is not in the document]
Comment (jurajmajerik) [reply posthog/api/test/test_scheduled_change.py:8]: I initially wanted to add them, but after reviewing _test_feature_flag_.py for reference, I noticed that we don't typically test these trivial CRUD cases (maybe with the exception of Create).

But happy to discuss and add more tests in a follow-up!

File: posthog/models/scheduled_change.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    id = models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name="ID")
+    record_id = models.IntegerField()
+    model_name: models.CharField = models.CharField(max_length=100, choices=AllowedModels.choices)
+    payload: models.JSONField = models.JSONField(default=dict)
+    scheduled_at: models.DateTimeField = models.DateTimeField(default=timezone.now)
+    executed_at: models.DateTimeField = models.DateTimeField(null=True, blank=True)
+    failure_reason = models.CharField(max_length=400, null=True, blank=True)
+
+    team: models.ForeignKey = models.ForeignKey("Team", on_delete=models.CASCADE)
+    created_at: models.DateTimeField = models.DateTimeField(auto_now_add=True)
```
[the remark this answers is not in the document]
Comment (jurajmajerik) [reply posthog/models/scheduled_change.py:18]: There's no editing UI in this initial iteration. IMO the Add & Delete is good enough for simple updates like "enable" and "add a release condition".

That said, the endpoint currently does accept a PUT/PATCH, so yes, will add this!

File: latest_migrations.manifest [hunk 3]
```diff
@@ -5,7 +5,7 @@ contenttypes: 0002_remove_content_type_name
 ee: 0015_add_verified_properties
 otp_static: 0002_throttling
 otp_totp: 0002_auto_20190420_0723
-posthog: 0373_externaldataschema
+posthog: 0375_auto_20231212_1128
```
[the remark this answers is not in the document]
Comment (jurajmajerik) [reply latest_migrations.manifest]: The CI test also forces me to name it properly :)

File: posthog/api/test/test_scheduled_change.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (liyiy) [reply posthog/api/test/test_scheduled_change.py:8]: Fair, probably makes more sense to add these updating/deletion tests for when it's connected to celery

**Review by neilkakkar — COMMENTED** [review]
👍

**Pull request merged**